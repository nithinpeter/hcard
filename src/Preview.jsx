import React from 'react';
import { commonPropTypes } from './common';

import imgPlaceholder from './avatar-placeholder.png';
import './Preview.scss';

const Preview = props => {
  return (
    <div className="preview">
      <div className="preview__content">
        <div className="preview__content__title">hCard preview</div>
        <Card {...props} />
      </div>
    </div>
  );
};

Preview.protoTypes = commonPropTypes;

const Card = props => (
  <div className="card">
    <div className="card__header">
      <h2 className="card__header__text">
        {props.givenName} {props.surname}
      </h2>
      <img
        className="card__header__img"
        src={props.avatar ? props.avatar : imgPlaceholder}
        alt="avatar"
      />
    </div>
    <div className="card__body">
      <div className="card__body__field">
        <span className="card__body__field__label">Email</span>
        <span className="card__body__field__value">{props.email}</span>
      </div>
      <div className="card__body__field">
        <span className="card__body__field__label">Phone</span>
        <span className="card__body__field__value">{props.phone}</span>
      </div>

      <div className="card__body__field">
        <span className="card__body__field__label">Address</span>
        <span className="card__body__field__value">
          {props.houseName} {props.street}
        </span>
      </div>
      <div className="card__body__field">
        <span className="card__body__field__label" />
        <span className="card__body__field__value">
          {[props.suburb, props.state].filter(val => !!val).join(', ')}
        </span>
      </div>

      <div className="card__body__field--double">
        <div className="card__body__field">
          <span className="card__body__field__label">Postcode</span>
          <span className="card__body__field__value">{props.postCode}</span>
        </div>

        <div className="card__body__field">
          <span className="card__body__field__label">Country</span>
          <span className="card__body__field__value">{props.country}</span>
        </div>
      </div>
    </div>
  </div>
);

Card.propTyes = commonPropTypes;

export default Preview;
