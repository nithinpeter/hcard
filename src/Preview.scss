@import './common.scss';

$preview-padding: 40px;
$card-padding: 20px;
$card-label-width: 70px;
$card-img-width: 80px;

.preview {
  padding: $preview-padding;
  flex: 1 0 50%;
  background: $gray-2;
  display: flex;
  justify-content: center;
  align-items: center;

  &__content {
    flex-grow: 1;
    max-width: 500px;

    &__title {
      text-transform: uppercase;
      text-align: right;
      padding-bottom: 10px;
      color: $gray-1;
    }
  }
}

.card {
  @include box-shadow();

  &__header {
    height: 100px;
    padding: $card-padding;
    background: $primary;
    color: $white;
    display: flex;
    align-items: flex-end;

    &__text {
      @include ellipsis;
      margin: 0 20px 0 0;
      flex-grow: 1;
      max-width: calc(
        100vw -
          (
            (2 * #{$preview-padding}) + (2 * #{$card-padding}) + #{$card-img-width}
          )
      );
    }

    &__img {
      @include box-shadow();
      margin-bottom: -40px;
      width: $card-img-width;
      height: 100px;
    }
  }

  &__body {
    padding: $card-padding $card-padding calc(#{$card-padding} + 10px);
    background: $white;

    &__field {
      padding-bottom: 3px;
      height: 40px;
      border-bottom: 1px solid gray;
      display: flex;
      align-items: flex-end;

      &__label {
        width: $card-label-width;
        text-transform: uppercase;
        font-size: 10px;
        color: $text-color;
      }

      &__value {
        max-width: calc(
          100vw -
            (
              (2 * #{$preview-padding}) + (2 * #{$card-padding}) + #{$card-label-width}
            )
        );
        @include ellipsis;
      }

      &--double {
        display: flex;

        .card__body__field {
          max-width: 50%;
          flex: 1 0 50%;
        }

        @media (max-width: 600px) {
          flex-direction: column;
          .card__body__field {
            max-width: 100%;
            flex: 1 0 100%;
          }
        }
      }
    }
  }
}
