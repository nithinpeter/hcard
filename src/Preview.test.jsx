import React from 'react';
import ReactDOMServer from 'react-dom/server';
import Preview from './Preview';

it('should render Preview', () => {
  const props = {
    givenName: 'Sam',
    surname: 'Fairfax',
    email: 'Sam@fairfax.com',
    phone: '012223344',
    houseName: '2',
    street: 'King st',
    suburb: 'Sydney',
    state: 'NSW',
    postCode: '2000',
    country: 'Australia',
  };
  expect(
    ReactDOMServer.renderToStaticMarkup(<Preview {...props} />)
  ).toMatchSnapshot();
});
