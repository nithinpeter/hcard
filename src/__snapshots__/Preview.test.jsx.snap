// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Preview 1`] = `"<div class=\\"preview\\"><div class=\\"preview__content\\"><div class=\\"preview__content__title\\">hCard preview</div><div class=\\"card\\"><div class=\\"card__header\\"><h2 class=\\"card__header__text\\">Sam Fairfax</h2><img class=\\"card__header__img\\" src=\\"avatar-placeholder.png\\" alt=\\"avatar\\"/></div><div class=\\"card__body\\"><div class=\\"card__body__field\\"><span class=\\"card__body__field__label\\">Email</span><span class=\\"card__body__field__value\\">Sam@fairfax.com</span></div><div class=\\"card__body__field\\"><span class=\\"card__body__field__label\\">Phone</span><span class=\\"card__body__field__value\\">012223344</span></div><div class=\\"card__body__field\\"><span class=\\"card__body__field__label\\">Address</span><span class=\\"card__body__field__value\\">2 King st</span></div><div class=\\"card__body__field\\"><span class=\\"card__body__field__label\\"></span><span class=\\"card__body__field__value\\">Sydney, NSW</span></div><div class=\\"card__body__field--double\\"><div class=\\"card__body__field\\"><span class=\\"card__body__field__label\\">Postcode</span><span class=\\"card__body__field__value\\">2000</span></div><div class=\\"card__body__field\\"><span class=\\"card__body__field__label\\">Country</span><span class=\\"card__body__field__value\\">Australia</span></div></div></div></div></div></div>"`;
